File: src/problem_1/problem_1.py
def run_1(lines: list[str]) -> None:
    total_sum = 0

    for line in lines:
        digits = [char for char in line if char.isdigit()]
        if digits:
            total_sum += int(f"{digits[0]}{digits[-1]}")

    print(f"Sum: {total_sum}")

File: src/problem_1/problem_1.rs
pub fn run_1(lines: &[String]) {
    let mut total_sum: i32 = 0;

    for line in lines {
        let digits: Vec<char> = line.chars().filter(|c| c.is_digit(10)).collect();

        if digits.is_empty() {
            continue;
        }

        let first_digit = digits.get(0).unwrap();
        let second_digit = digits.last().unwrap();

        let two_digit_number = format!("{}{}", first_digit, second_digit)
            .parse::<i32>()
            .unwrap_or(0);
        total_sum += two_digit_number;
    }

    println!("Sum: {}", total_sum);
}
